src/main/default/classes/Manager.cls

```diff
@@ -52,134 +70,100 @@ global class Manager {
     global void manage() {
-        if (triggerConfig != null && !triggerConfig.enabled) {
+        if (!triggerConfig.enabled) {
             return;
         }
 
-        List<Handler> handlers = eventHandlerMapping.get(context.operationType);
+        List<HandlerConfig> handlers = eventHandlerMapping.get(context.operationType);
 
         if (handlers != null && !handlers.isEmpty()) {
-            for (Handler h : handlers) {
-                System.debug('Running handler: ' + getHandlerName(h));
-                if (h instanceof BaseHandler) {
-                    ((BaseHandler) h).context = this.context;
+            for (HandlerConfig config : handlers) {
+                System.debug('Running handler: ' + config.className);
+                if (config.handler instanceof BaseHandler) {
+                    ((BaseHandler) config.handler).context = this.context;
                 }
-                if (isDebugOnly(h)) {
-                    try {
-                        h.handle();
-                    } catch (Exception e) {
-                        System.debug(JSON.serializePretty(e));
-                    }
-                } else {
-                    h.handle();
+                Savepoint sp = Database.setSavePoint();
+                try {
+                    config.handler.handle();
+                } catch (Exception e) {
+                    exceptionManager.manage(e, context, config, sp);
                 }
             }
         }
     }
 
-    private void doAutomaticBinding() {
-        if (triggerConfig == null || !triggerConfig.enabled) {
+    /**
+     * Handles binding of configuration based handles
+     */
+    private void bindDynamicConfiguration() {
+        if (!triggerConfig.enabled) {
             return;
         }
         for (String handlerName : triggerConfig.handlerOrder) {
             System.debug('Auto binding: ' + handlerName);
-            TriggerConfigService.HandlerConfig handlerConfig = triggerConfig.handlers.get(handlerName);
+            HandlerConfig handlerConfig = triggerConfig.handlers.get(handlerName);
             // do not bind if auto bind is off or if handler is disabled
             if (!handlerConfig.enabled) {
                 continue;
             }
-            Type t = getType(handlerConfig.className);
-            if (t == null) {
-                System.debug('Class does not exist: ' + handlerConfig.className);
-                continue;
-            }
-            Handler handlerClass;
-            try {
-                handlerClass = (Handler) t.newInstance();
-            } catch (TypeException te) {
-                System.debug(
-                    'Cannot bind invalid class ' +
-                    handlerConfig.className +
-                    '. Make sure your handler implements the Handler'
-                );
+            Handler handlerClass = constructHandler(handlerConfig);
+            if (handlerClass == null) {
                 continue;
             }
 
```

Comment:
Do we actually need to dynamically bind all events?  

I think we JUST need to bind the current `context.TriggerOperation`, but maybe I'm missing something